### ISSUE TITLE
DL4j BarnesHutTsne Workspace-Panic

### ISSUE THREAD

**Issue by G4mble** [deeplearning4j/deeplearning4j#5292]
I'm using the TSNE-example from the deeplearning4j example project.
I tried using three different inputs:
(1) Text-file with 6k words (Format: one word per line; wordName wordVector)
(2) Text-file with ~18k words (Format: one word per line; wordName wordVector)
(3) Word2Vec-Model containing ~18k words in the vocabulary

For inputs (2) and (3) I get the following exception [full output included]
https://gist.github.com/G4mble/a9f35e78fd9499ccf2970269a78e99b6

Input (1) works fine; no exceptions, output as expected.

The code for input (2) and (3) can be found here [for input (2) the code in lines 42-49 is used; for input (3) the code in lines 51-54 is used]:
https://gist.github.com/G4mble/5f370e30630ded9737e7bd71e9e6625a

The code already contains part of my attempt to tackle the issue. In line 33 I tried the try-catch solution suggested here: https://nd4j.org/userguide#workspaces-panic
I also tried using ".detach()" on the INDArray in line 54, this did not fix the error.

The pom.xml can be found here:
https://gist.github.com/G4mble/af6bd35d1b685bd902833585cd2c43c5

OS: Win10 Pro-64bit German
IDE: IntelliJIdea Community 2018.1.2x64
Backend: CPU

**Comment by AlexDBlack** [comment]
Was fixed earlier here: https://github.com/deeplearning4j/deeplearning4j/pull/5971
